giftcertificate/service/src/main/java/com/epam/esm/service/impl/TagServiceImpl.java

```diff
@@ -119,10 +128,41 @@ public void delete(long id) throws IdNotExistServiceException {
      *
      * @return list of TagDto
      */
+    @Transactional
     @Override
-    public List<TagDto> findAll() {
-        List<Tag> tags = tagDAO.findAll();
+    public List<TagDto> findAll(Integer page, Integer size) throws PaginationException {
+        page = checkPage(page);
+        size = checkSizePage(size);
+        List<Tag> tags = tagDAO.findAll(page, size);
         return tags.stream().map(tag -> modelMapper.map(tag, TagDto.class)).collect(Collectors.toList());
     }
 
+    @Override
+    public TagDto getWidelyUsedByUserTagWithHighestCost(long userId) throws IdNotExistServiceException {
+        long idTag;
+        try {
+            idTag = tagDAO.getIdWidelyUsedByUserTagWithHighestCost(userId);
+        } catch (NullPointerException e) {
+            throw new IdNotExistServiceException("User with id = " + userId + " is not exist in DB or has not any orders");
+        }
+        TagDto tagDto = modelMapper.map(tagDAO.read(idTag), TagDto.class);
```

Comment:
you can just return it outright instead of declaring variable and then returning it